### ISSUE TITLE
[Platform] Android phone failed to connect to a FireTV device

### ISSUE THREAD

**Issue by Hiwensen** [project-chip/connectedhomeip#34224]
### Reproduction steps

1. Run the tv-casting-app on an Android Pixel 7a device running Android Version 14
2. Run the tv-content-app on a FireTV Stick 4K Max device with Fire OS 7.6.7.3
3. Open the tv-casting-app on the phone, wait for the discovery
4. Click the FireTV device discovered in the list and fail to connect to the device.
Please check the screenshot and logs.

<img width="1549" alt="Screenshot 2024-07-08 at 11 20 18" src="https://github.com/project-chip/connectedhomeip/assets/14145498/e9c8bc73-0d89-45d0-9ec6-41760e610b88">

[connect_failed_logcat.txt](https://github.com/user-attachments/files/16122430/connect_failed_logcat.txt)

### Platform

android

### Platform Version(s)

N/A

### Type

Manually tested with SDK

### Anything else?

Question:
Should I run the tv-platform app on the FireTV device to make the connection work?

**Comment by Hiwensen** [comment]
Hi @sharadb-amazon, Do we have any plan to fix this?

**Comment by Hiwensen** [comment]
I updated to the latest code with commitId `63182c41e71bbbe` and tried connecting an Android phone to a FireTV device, but the connection still failed. 
Below are the error logs and screenshots:
<img width="2514" alt="Screenshot 2024-11-27 at 14 41 10" src="https://github.com/user-attachments/assets/91b92fbc-d598-4670-8b51-54245d9ed30d">

[matter_connect_log.txt](https://github.com/user-attachments/files/17945449/matter_connect_log.txt)

**Comment by pgregorr-amazon** [comment]
@lazarkov please help answer @Hiwensen questions above regarding the tv-platform app compatibility with different platforms.

**Comment by lazarkov** [comment]
> 1. Should I run the tv-platform app on the FireTV device to make the connection work?
> 2. Currently, the tv-casting app can only discover FireTV devices running FireOS 7 and above.
> It can't discover the other FireTV, AndroidTV, and Roku devices, is this the expected behaviour?


1. No. It's sufficient that you install your own content app and that should help you with the connection.
2. Yes. That's the expected behaviour. However once AndroidTV and Roku start running Matter on their production OS then those device will be visible as well.

**Comment by Hiwensen** [comment]
@sharadb-amazon Thank you for your reply!
> While reproducing this error, can you upload the logs from the Fire TV side to help us understand why we got the Invalid PASE parameter?

Here are the logs from the FireTV side.
[matter_content_app_logcat.txt](https://github.com/user-attachments/files/18056275/matter_content_app_logcat.txt)

>Note that when using the tv-casting-app, we've usually tested against the sample Linux/Android tv-app.

I will test on a ATV device and get back to you later.

**Comment by Hiwensen** [comment]
@pgregorr-amazon 
> The next step should be accepting the connection request on the tv-app/Fire TV. This does not happen automatically on the first connection attempt.

I remember that once during the connection, a PIN code input dialog appeared on FireTV. After entering the PIN code, I didn’t see any response. Later, when connecting again, the dialog no longer appeared.

Below are the display pages of the casting app on the Android phone and the content app on FireTV.
![matter_android](https://github.com/user-attachments/assets/4a59790a-5d69-4e67-b6f4-fbecc1d5009b)
![ftv](https://github.com/user-attachments/assets/c9812b57-1d0c-4861-93e2-b6e7a72ec866)

**Comment by Hiwensen** [comment]
@sharadb-amazon 
> Note that when using the tv-casting-app, we've usually tested against the sample Linux/Android tv-app.

Unfortunately, after running the TV-platform and TV-content apps on an AndroidTV mailbox running Android 9 and an AndroidTV mailbox running Android 11, the phone's casting app cannot discover either of them.
I can share the logs from the TV and phone here. For now, I think we can focus on the connection on FireTV.

**Comment by pgregorr-amazon** [comment]
@Hiwensen can you please confirm that we can close this issue?
Resolution:
Casting from the example tv-casting-app to a FireTV requires the developer's Amazon CustomerID to be allow listed for DAC Attestation step in the cloud service.

**Comment by Hiwensen** [comment]
> @Hiwensen can you please confirm that we can close this issue? Resolution: Casting from the example tv-casting-app to a FireTV requires the developer's CustomerID to be allow listed for DAC Attestation step in the cloud service.

Yes, we can close this issue, thank you for your help with the connection and adding the CustomerID to be the allow listed.

**Comment by xqb1** [comment]
Have you solved this problem? I am also facing this problem now.

**Comment by xqb1** [comment]
> [@Hiwensen](https://github.com/Hiwensen)您能确认我们可以解决这个问题吗？ 解决方案： 从示例 tv-casting-app 投射到 FireTV 需要开发者的客户 ID 被列入云服务中 DAC 认证步骤的允许名单。

Hello, I would like to ask how to add the customer ID to the DAC permission list?
